### ISSUE TITLE
Default values not populated if query contains a `select` statement

### ISSUE THREAD

**Issue by EgoAleSum** [LearnBoost/mongoose#2715]
My schema for the "Pool" collection contains some default values.

**Code**
````javascript
Pool
	.findById(poolId)
	.select('configuration')
	.exec(function(err, pool) {
		// ..
	})
````

**Expected result**
`pool` should have empty values (in `pool.configuration`) populated with defaults from the schema

**Actual result**
Defaults from the schema are not used and empty values are not populated

**Troubleshooting**
If using:
````javascript
Pool
	.findById(poolId)
	//.select('configuration')
	.exec(function(err, pool) {
		// ..
	})
````
(note the `.select` commented out)
Then defaults are used to populate empty values.

**Comment by vkarpov15** [comment]
This looks like a duplicate #2629 #2670 etc. Can you verify if this is the case, and, if so, if upgrading to the recently released v3.8.24 fixes your issue?

**Comment by EgoAleSum** [comment]
You're correct: I was still using 3.8.23. Thanks for fixing that already
